src/pages/_DevSitePage.jsx

```diff
@@ -0,0 +1,150 @@
+// import IconLocationPin from 'terra-icon/lib/icon/IconLocationPin';
+import IconStartPresenting from 'terra-icon/lib/icon/IconStartPresenting';
+import { useLocation, useHistory, useRouteMatch } from 'react-router-dom';
+
+import PageErrorBoundary from './_DevSitePageErrorBoundary';
+import LoadingOverlay from './_LoadingOverlay';
+import ContentLoadedContainer from './_ContentLoaded';
+import AppSettingsContext from '../site/_AppSettingsContext';
+import ContentSettingsMenu from './_ContentSettingsMenu';
+// import ComponentToolbar from './_ComponentToolbar';
+import NotFoundPage from './_NotFoundPage';
+import { contentImportsPropType, pageContentConfigPropType } from '../site/siteConfigPropTypes';
+
+const propTypes = {
+  /**
+   * Config describing the page
+   */
+  pageContentConfig: pageContentConfigPropType.isRequired,
+
+  /**
+   * Function called to request closing the modal
+   */
+  contentImports: contentImportsPropType.isRequired,
+};
+
+const DevSitePage = ({ pageContentConfig, contentImports }) => {
+  const location = useLocation();
+  const history = useHistory();
+  const isRaw = useRouteMatch('/raw');
+  const isHome = useRouteMatch('/home');
+  const localeButtonRef = React.useRef();
+  const themeButtonRef = React.useRef();
+  const [showThemeMenu, setShowThemeMenu] = React.useState(false);
+  const [showLocale, setShowLocale] = React.useState(false);
+  const appSettings = React.useContext(AppSettingsContext);
+  const { isActive } = React.useContext(NavigationItemContext);
+
+  if (!isActive) {
+    return null;
+  }
+
+  const pathname = isRaw ? location.pathname.substring(4) : location.pathname;
+  // const hasThemes = appSettings.themes && appSettings.themes.length > 1;
+  // const hasLocales = appSettings.locales && appSettings.locales.length > 1;
+  const ContentComponent = contentImports[pathname];
+
+  // Last Chance 404
+  if (!ContentComponent) {
+    return <NotFoundPage />;
+  }
+
+  const onChangeTheme = (theme) => {
+    setShowThemeMenu(false);
+    appSettings.onUpdate({ theme });
+  };
+
+  const onChangeLocale = (locale) => {
+    setShowLocale(false);
+    appSettings.onUpdate({ locale });
+  };
+
+  const pageActions = [
+    {
+      key: 'raw',
+      label: 'RawToggle',
+      icon: <IconStartPresenting />,
+      onSelect: () => {
+        history.push(`/raw${location.pathname}`);
+      },
+    },
+  ];
+
+  // if (hasThemes) {
+  //   pageActions.push({
+  //     key: 'theme',
+  //     label: 'Theme',
+  //     icon: <IconTreemap />,
+  //     onSelect: () => { setShowThemeMenu(true); },
+  //     buttonRefCallback: (ref) => { themeButtonRef.current = ref; },
+  //   });
+  // }
+
+  // if (hasLocales) {
+  //   pageActions.push({
+  //     key: 'locale',
+  //     label: 'Locale',
+  //     icon: <IconLocationPin />,
+  //     onSelect: () => { setShowLocale(true); },
+  //     buttonRefCallback: (ref) => { localeButtonRef.current = ref; },
+  //   });
+  // }
```

Comment:
we may re-enable these depending on where we land on quick actions to change the theme and locale.